contextproject-ms3/src/main/java/contextproject/loaders/FolderLoader.java

```diff
@@ -0,0 +1,72 @@
+package contextproject.loaders;
+
+import java.io.File;
+import java.util.ArrayList;
+
+/**
+ * Class to read all the mp3 files from a given folder.
+ *
+ */
+public class FolderLoader implements PlaylistLoader {
+
+  private File folder;
+  private ArrayList<String> list;
+
+  /**
+   * Constructor.
+   * 
+   * @param location
+   *          location of the folder.
+   */
+  public FolderLoader(String location) {
+    folder = new File(location);
+    list = new ArrayList<String>();
+  }
+
+  /**
+   * Method to add all mp3 files from the folder and folders within this folder to the list.
+   * 
+   * @param directory
+   *          location of the folder.
+   */
+  public void addToList(File directory) {
+    try {
+      File[] allfiles = directory.listFiles();
+
+      for (int i = 0; i < allfiles.length; i++) {
+        if (allfiles[i].length() >= 4 && allfiles[i].toString().endsWith(".mp3")) {
+          list.add(allfiles[i].toString());
+        } else if (allfiles[i].isDirectory()) {
+          FolderLoader loader = new FolderLoader(allfiles[i].toString());
+          File submap = new File(allfiles[i].toString());
+          loader.addToList(submap);
+          list.addAll(loader.getList());
```

Comment:
This is also unnecessary complex. You're creating a FolderLoader only to call its addToList method. The entire line 40-43 can probably be replaced by a call to addToList directly. Then you can make addToList private as well.